packages/beacon-node/src/api/impl/beacon/pool/index.ts

```diff
@@ -272,7 +272,7 @@ export function getBeaconPoolApi({
               chain.syncCommitteeMessagePool.add(subnet, signature, indexInSubcommittee);
 
               // Cheap de-duplication code to avoid using a Set. indexesInCommittee is always sorted
-              if (subnets.length === 0 || subnets[subnets.length - 1] !== subnet) {
+              if (subnets.length === 0 || subnets.at(-1) !== subnet) {
```

Comment:
so `-1` seems to be last item in the array, this is very confusing syntax

Reply:
> arr.slice(-2); returns last two items.

yeah I know, somehow it makes more sense to me here, it's probably just a matter of getting used to it, if that's what needed to make array access more type safe I am all for it

Reply:
Yes `.at` give type-safety compared to index access of the array. In case of under index or over index access the array returns `undefined` but the type for index access is always without undefined. Which is resolved with `.at`, so you must check if what you accessed is available or not.

Reply:
But that also need a careful review because `if(!arr.at(-1))` such usages may fail if last value exists but is some falsy.